lib/ui/common/webview.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:farmsmart_flutter/ui/common/ContextualAppBar.dart';
+import 'package:flutter/material.dart';
+import 'package:webview_flutter/webview_flutter.dart' as FlutterWebView;
+
+class _Constants {
+  static final loadingIndex = 0;
+  static final webViewIndex = 1;
+}
+
+class WebView extends StatefulWidget {
+  final url;
+
+  const WebView({Key key, this.url}) : super(key: key);
+
+  @override
+  _WebViewState createState() => _WebViewState();
+}
+
+class _WebViewState extends State<WebView> {
+  int _stackToView;
+
+  @override
+  void initState() {
+    _stackToView = _Constants.loadingIndex;
+    super.initState();
+  }
+
+  void _handlePageLoaded(String value) {
+    setState(() {
+      _stackToView = _Constants.webViewIndex;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return IndexedStack(
```

Comment:
what's the thinking on using a stack?

Reply:
I think is using a stack for showing the loading spinner or the webView, depending on the index given through `_stackToView`